Miftakhunnurudin

Miftakhunnurudin opened the pull request:
.env

DB_USERNAME_TEST = postgres
DB_PASSWORD_TEST = postgres
DB_NAME_TEST = ecommerce_test
DB_HOST_TEST = 127.0.0.1
DB_DIALECT_TEST = postgres

DB_USERNAME_DEVELOPMENT = postgres
DB_PASSWORD_DEVELOPMENT = postgres
DB_NAME_DEVELOPMENT = ecommerce_development
DB_HOST_DEVELOPMENT = 127.0.0.1
DB_DIALECT_DEVELOPMENT = postgres

SECRET_KEY = myhobbyissleeping

server = https://server-ecommerce-nur.herokuapp.com/

Miftakhunnurudin committed:
add server:
login

add test:
login user

Miftakhunnurudin committed:
fix the issue according the test expectation

Miftakhunnurudin committed:
add:
-apis for create, get, update and delete products
-success test for create,update and delete products
-fail test for create,update and delete products

Miftakhunnurudin committed:
add:
- cors in app.js
- add script to runs server in development env
- err handler in middlewares
- routes
- seeds program for admin account

Miftakhunnurudin committed:
changes:
- rename test file from login.tes.js to all.test.js
- change authadmin to authorization and authentication
- changes routes middlewares for auth
- some production setting in package.json

additions:
- api documentation
- seeding database for product

Miftakhunnurudin committed:
addition:
- cartController
- Cart model and migration
- register at authController
- cart routes
- add routes for update stock of product that can be accessed by customer

changes:
- change login from only admin that can login, now customer can
- assotiation product and user

Sursev07 commented on controllers/cartController.js:
```diff
@@ -0,0 +1,62 @@
+class CartController {
+    static async create(req,res,next){
+        try {
+            let data = req.body
```
stock nya?

Sursev07 replied:
kalau stock lebih kecil dari quantity cart?

Sursev07 commented on controllers/cartController.js:
```diff
@@ -0,0 +1,62 @@
+    static async update(req,res,next){
+        try {
+            let data = req.body
+            let result = await Cart.update(data,{
```
stock product nya ? kalau lebih kecil dari quantity cart?

Sursev07 commented on models/cart.js:
```diff
@@ -0,0 +1,31 @@
+      // define association here
+    }
+  };
+  Cart.init({
```
validasi?

Sursev07 commented on routes/cart.js:
```diff
@@ -0,0 +1,9 @@
+const CartController = require('../controllers/cartController')
+
+router.get('/',CartController.get)
+router.put('/:id',CartController.update)
```
validasi ga bisa update atau delete cart milik user lain?

Miftakhunnurudin committed:
addition:
- cart controller {crud}
- history {cr}
- routes history

changes:
- some setting in migrations
- debugging auth.js (salah penamaan variabel)
- fixing the routes
